lxd/storage/drivers/drivers_lvm.go

```diff
@@ -0,0 +1,561 @@
+		if err != nil {
+			d.logger.Warn("Failed to destroy the physical volume for the lvm storage pool", log.Ctx{"err": err})
+		}
+		d.logger.Debug("Physical volume removed", log.Ctx{"pv_name": loopFile.Name()})
+
+		// Set LO_FLAGS_AUTOCLEAR before removing the loop file otherwise we will get EBADF.
+		err = SetAutoclearOnLoopDev(int(loopFile.Fd()))
+		if err != nil {
+			d.logger.Warn("Failed to set LO_FLAGS_AUTOCLEAR on loop device, manual cleanup needed", log.Ctx{"err": err})
+		}
+
+		err = loopFile.Close()
+		if err != nil {
+			return err
+		}
+
+		// This is a loop file so deconfigure the associated loop device.
+		err = os.Remove(d.config["source"])
+		if err != nil {
+			return errors.Wrapf(err, "Error removing LVM pool loop file '%s'", d.config["source"])
+		}
+		d.logger.Debug("Physical loop file removed", log.Ctx{"file_name": d.config["source"]})
+	}
+
+	// Wipe everything in the storage pool directory.
+	err = wipeDirectory(GetPoolMountPath(d.name))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *lvm) Validate(config map[string]string) error {
+	return nil
+}
+
+// Update updates the storage pool settings.
+func (d *lvm) Update(changedConfig map[string]string) error {
+	if changedConfig["lvm.vg_name"] != "" {
+		err := d.renameVolumeGroup(d.config["lvm.vg_name"], changedConfig["lvm.vg_name"])
+		if err != nil {
+			return err
+		}
+	}
+
+	if changedConfig["lvm.thinpool_name"] != "" {
+		err := d.renameThinpool(d.config["lvm.vg_name"], d.config["lvm.thinpool_name"], changedConfig["lvm.thinpool_name"])
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Mount mounts the storage pool (this does nothing for external LVM pools, but for loopback image
+// LVM pools this creates a loop device).
+func (d *lvm) Mount() (bool, error) {
+	// Open the loop file if the LVM device doesn't exist yet and the source points to a file.
+	if !shared.IsDir(fmt.Sprintf("/dev/%s", d.config["lvm.vg_name"])) && filepath.IsAbs(d.config["source"]) && !shared.IsBlockdevPath(d.config["source"]) {
+		loopFile, err := d.openLoopFile(d.config["source"])
+		if err != nil {
+			return false, err
+		}
+		defer loopFile.Close()
+
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// Unmount unmounts the storage pool (this does nothing for external LVM pools, but for loopback
+// image LVM pools this closes the loop device handle if needed).
+func (d *lvm) Unmount() (bool, error) {
+	if filepath.IsAbs(d.config["source"]) && !shared.IsBlockdevPath(d.config["source"]) {
+		loopFile, err := d.openLoopFile(d.config["source"])
+		if err != nil {
+			return false, err
+		}
+
+		// Set LO_FLAGS_AUTOCLEAR before removing the loop file otherwise we will get EBADF.
+		err = SetAutoclearOnLoopDev(int(loopFile.Fd()))
+		if err != nil {
+			d.logger.Warn("Failed to set LO_FLAGS_AUTOCLEAR on loop device, manual cleanup needed", log.Ctx{"err": err})
+		}
+
+		err = loopFile.Close()
+		if err != nil {
+			return false, err
+		}
```

Comment:
This entire chunk would likely be better handled by `releaseLoopDev` similar to what's done in btrfs.

Reply:
@stgraber ah OK will use releaseLoopDev instead.